src/libs/Trie/TrieNode.ts

```diff
@@ -1,4 +1,13 @@
-type MetaData = Record<string, unknown>;
+type Suggestion = {
+    code: string;
+    types?: string[];
+    name?: string;
+};
+
+type MetaData = {
```

Comment:
`Trie` and `TrieNode` were typed with generic MetaData in mind. So instead of overwriting it here define `EmojiMetaData` inside of `EmojiTrie` file and pass it as a generic argument:

```ts
// src/libs/EmojiTrie.ts
type Suggestion = {
    code: string;
    types?: string[];
    name?: string;
};

type EmojiMetaData = {
    suggestions?: Suggestion[];
};

Trie<MetaData> -> Trie<EmojiMetaData>
```

Reply:
Updated